### PR TITLE
catalog deeplinks

### PR DESCRIPTION
**Pull request by grzegorzdec**

### CODE REVIEW

File: global/globalVariables.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export var initialDeeplink = undefined
```
Comment (pleszkiewicz) [inline global/globalVariables.js]: Do we need to export `initialDeeplink`?

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -33,6 +33,12 @@
             <action android:name="android.intent.action.MAIN" />
             <category android:name="android.intent.category.LAUNCHER" />
         </intent-filter>
+        <intent-filter android:label="filter_react_native">
+            <action android:name="android.intent.action.VIEW" />
+            <category android:name="android.intent.category.DEFAULT" />
+            <category android:name="android.intent.category.BROWSABLE" />
+            <data android:scheme="catalog" android:host="scene" /> // A
```
Comment (pleszkiewicz) [inline android/app/src/main/AndroidManifest.xml]: What does `// A` mean? Do we need it?

Comment (pleszkiewicz) [reply android/app/src/main/AndroidManifest.xml]: What's the format of the final deeplink? `catalog://scene=3` ? `catalog://scene/3` ?

File: proxy_mobile/react-native/ReactNativeApp.js [hunk 3]
```diff
@@ -2,12 +2,17 @@ import React from 'react';
...
 
 export default class ReactNativeApp extends React.Component {
-  
+
   constructor(props) {
     super(props);
-    this.state = { planeDetection: true} ;
+    this.state = { planeDetection: true };
+    
+    if(props.initialUrl != null) {
```
Comment (pleszkiewicz) [inline proxy_mobile/react-native/ReactNativeApp.js]: Do we need to check `initialUrl` agains null/undefined?
BTW In .js we should use `!==/===` for comparison ([SO link](https://stackoverflow.com/questions/2778901/how-to-compare-variables-to-undefined-if-i-don-t-know-whether-they-exist)).

File: global/globalVariables.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+deeplinkSet = () => undefined
+
+export function registerOnDeeplinkSet(onDeeplinkSet) {
+    deeplinkSet = onDeeplinkSet
+}
+
+export function setInitialDeeplink(deeplink) {
+    initialDeeplink = deeplink
+    if (deeplinkSet != undefined) {
```
Comment (pleszkiewicz) [inline global/globalVariables.js]: `!=` -> `!==`

File: global/globalVariables.js (hunk 1 above)
Comment (grzegorzdec) [reply global/globalVariables.js]: probably not, good catch

File: android/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (grzegorzdec) [reply android/app/src/main/AndroidManifest.xml]: `catalog://scene/3`, I added comment in `catalog_app.js`

File: proxy_mobile/react-native/ReactNativeApp.js (hunk 3 above)
Comment (grzegorzdec) [reply proxy_mobile/react-native/ReactNativeApp.js]: in my opinion, there is no need to notify observers if there is no `initialUrl`

**Pull request merged**